src/call.ts

```diff
@@ -118,13 +118,9 @@ export class OngoingCallPromise extends OngoingCall {
       rawResponse?: RawResponseType
     ) => {
       if (err) {
-        if (err.metadata) {
-          const decoder = new GoogleErrorDecoder();
-          err.statusDetails = decoder.decodeRpcStatusDetails(
-            err.metadata.get('grpc-status-details-bin') as Buffer[]
-          );
-        }
-        rejectCallback(err);
+        const decoder = new GoogleErrorDecoder();
+        const decodedErr = decoder.decodeMetadata(err);
```

Comment:
Let's put this block in try-catch and if an exception happens, call `rejectCallback` wit hthe original `err`.

Reply:
Done